python/ray/tune/impl/tuner_internal.py

```diff
@@ -218,6 +233,18 @@ def _restore_from_path_or_uri(
             tuner = pickle.load(fp)
             self.__dict__.update(tuner.__dict__)
 
+        # Re-wrap the trainable with newly-specified parameters, if provided
+        # If not, throw an error when trying to fit
+        if hasattr(trainable, "_unwrapped_trainable_and_params"):
+            (
+                unwrapped_trainable,
+                attached_params,
+            ) = trainable._unwrapped_trainable_and_params
+            if not parameters or set(parameters.keys()) != set(attached_params):
+                self._missing_trainable_parameters = attached_params
+            else:
+                trainable = tune.with_parameters(unwrapped_trainable, **parameters)
```

Comment:
This is a nit and feel free to ignore it, but it would be nice to separate this out into a method